packages/vscode-extension/src/devices/DeviceManager.ts

```diff
@@ -125,11 +128,11 @@ export class DeviceManager implements DeviceManagerInterface {
     return getAvailableIosRuntimes();
   }
 
-  public async listAllDevices() {
+  public async listAllDevices(forceReload?: boolean) {
     const devices = extensionContext.globalState.get(DEVICE_LIST_CACHE_KEY) as
       | DeviceInfo[]
       | undefined;
-    if (devices) {
+    if (devices && !forceReload) {
```

Comment:
this isn't accurate as we *always* load the devices. The difference is that we return the updated list or let it update asynchronously. I'm a bit hesitant about this change as from what I remember, we added device caching as it was slowing down the initial startup significantly. As it isn't documented in PR description I wonder why you're changing this behavior?

Reply:
When user had an iPhone selected when they last used IDE, but something happend to xcode (e.g. user started using beta but not updated the `xcode-select`)  application would select an iPhone anyway as it is the most recent device and would get stuck on loading screen. 

What dou you think about `forceLoad` instead of `forceReload`

Reply:
I think we should optimize for the most common case in which people don't change their xcode location. If there are cases when it happens, I understand they will get an error when trying to load the simulator, and by that time we will have the list refetched. If that's not the case we should adjust the logic such that the worst thing that can happen if you mess up the simjulator cache or xcode is that you get an error loading the simulator instead of slowing down all the time to interact for everyone else.